bentoml/onnx.py

```diff
@@ -274,28 +277,34 @@ def _setup(self) -> None:  # type: ignore[override]
             session_options=self._session_options,
             model_store=self._model_store,
         )
-        self._infer_func = getattr(self._model, "run")
+        raw_infer_func = getattr(self._model, "run")
+        self._infer_func = functools.partial(raw_infer_func, **self._partial_kwargs)
 
     def _run_batch(
         self,
-        *args: t.Union["np.ndarray", "pd.DataFrame"],
-        **kwargs: t.Any,
+        *args: t.Union[np.ndarray, "pd.DataFrame"],
     ) -> t.Any:
-        params = Params[t.Union["np.ndarray", "pd.DataFrame"]](*args, **kwargs)
-        if isinstance(params.sample, np.ndarray):
-            params = params.map(lambda i: i.astype(np.float32))
-        elif isinstance(params.sample, pd.DataFrame):
-            params = params.map(lambda i: i.to_numpy())
-        else:
-            raise TypeError(
-                f"`_run_batch` of {self.__class__.__name__} only takes "
-                "`numpy.ndarray` or `pd.DataFrame` as input parameters"
-            )
+        params = Params[t.Union[np.ndarray, "pd.DataFrame"]](*args)
+
+        def _mapping(item) -> t.Any:
+            if isinstance(item, np.ndarray):
+                item = item.astype(np.float32)
+            elif isinstance(item, pd.DataFrame):
+                item = item.to_numpy()
+            else:
+                raise TypeError(
+                    f"`_run_batch` of {self.__class__.__name__} only takes "
+                    "`numpy.ndarray` or `pd.DataFrame` as input parameters"
+                )
```

Comment:
should we support torch.Tensor? @aarnphm

Reply:
sure, but I think we might need to do lazy loading for torch and tensorflow

Reply:
done